.github/workflows/test-go.yml

```
@@ -365,6 +365,7 @@ jobs:
           export VAULT_TEST_LOG_DIR
           mkdir -p "$VAULT_TEST_LOG_DIR"
           
+          unset VAULT_ADDR
           
           # shellcheck disable=SC2086 # can't quote RERUN_FAILS
           GOARCH=${{ inputs.go-arch }} \
```

I don't actually know how VAULT_ADDR gets set, nor do I really understand the lifecycle of env vars in actions.  So to me, the safest option is to do like we do for GOARCH, i.e, set the variable only for the gotestsum invocation:

```suggestion
          GOARCH=${{ inputs.go-arch }} VAULT_ADDR= \
```